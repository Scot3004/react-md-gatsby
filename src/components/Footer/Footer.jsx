import React, { Component } from "react";
import Button from "react-md/lib/Buttons";
import Link from "gatsby-link";
import UserLinks from "../UserLinks/UserLinks";
import "./Footer.scss";

class Footer extends Component {
  render() {
    const { userLinks, config } = this.props;
    const { copyright, fixedFooter, siteRss } = config;
    return (
      <footer className={fixedFooter ? "footer footer-fixed" : "footer"}>
        {userLinks ? <UserLinks config={config} labeled /> : null}
        <div className="notice-container">
          <div className="copyright">
            <h4>{copyright}</h4>
          </div>

          <div className="rss">
            <Link to={siteRss}>
              <Button flat primary swapTheming iconClassName="fa fa-rss">
                Subscribe
              </Button>
            </Link>
          </div>
          <div className="based-on">
            <h4>
              Based on{" "}
              <a href="https://github.com/Vagr9K/gatsby-material-starter">
                Gatsby Material Starter
              </a>.
            </h4>
          </div>
        </div>
      </footer>
    );
  }
}

Footer.defaultProps = {
  config: {}
}

export default Footer;
