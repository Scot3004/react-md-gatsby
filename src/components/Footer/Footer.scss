@import '../../layouts/theme.scss';

.footer {
  color: $md-white-base;
  justify-content: center;
  align-content: center;
  background-color: $md-primary-color;
  padding: 10px 5px 5px;

  .notice-container {
    display: flex;
    align-content: center;
    align-items: center;
    margin-top: 1rem;

    .copyright {
      display: flex;
      flex: 1;
      justify-content: flex-start;
      padding: 0 1rem;

      @media (max-width: 640px - 1px) {
        justify-content: center;
      }
    }

    .rss {
      display: flex;

      @media (max-width: 640px - 1px) {
        display: none;
      }

      flex: 1;
      justify-content: center;
    }

    .based-on {
      display: flex;

      @media (max-width: 640px - 1px) {
        display: none;
      }

      flex: 1;
      justify-content: flex-end;

      a {
        padding-left: 0.4em;
      }
    }
  }
}

.footer .md-btn,
.footer-link {
  &:link,
  &:visited {
    color: $md-white-base;
  }
}

.footer-fixed {
  position: fixed;
  left: 0;
  bottom: 0;
  right: 0;
  z-index: 15;
}
