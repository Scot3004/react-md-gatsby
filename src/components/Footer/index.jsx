import React, { Component } from "react";
import Button from "react-md/lib/Buttons";
import Link from "gatsby-link";
import UserLinks from "../UserLinks";
import "./Footer.scss";

class Footer extends Component {
  render() {
    const { userLinks, config } = this.props;
    const { copyright, fixedFooter, siteRss } = config;
    return (
      <footer className={fixedFooter ? "footer footer-fixed" : "footer"}>
        {userLinks ? <UserLinks config={config} labeled /> : null}
        <div className="notice-container">
          <div className="copyright">
            {copyright}
          </div>
          <div className="rss">
            <Link to={siteRss}>
              <Button flat primary swapTheming iconClassName="fa fa-rss">
                Subscribe
              </Button>
            </Link>
          </div>
          <div className="based-on">
            Based on
            <a
              className="footer-link"
              href="https://github.com/Vagr9K/gatsby-material-starter"
            >
              Gatsby Material Starter
            </a>.
          </div>
        </div>
      </footer>
    );
  }
}

Footer.defaultProps = {
  config: {}
}

export default Footer;
