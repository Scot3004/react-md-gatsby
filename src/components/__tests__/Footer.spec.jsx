import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Footer from '../Footer';


it('renders correctly userlinks', () => {
  const config = {
    userLinks: "UserLinks",
    siteRss: "rss"
  };
  const tree = shallow(<Footer userLinks config={config} />);
  expect(toJson(tree)).toMatchSnapshot();
});

it('renders correctly fixedFooter', () => {
  const config = {
    fixedFooter: true,
    siteRss: "rss"
  };
  const tree = shallow(<Footer config={config} />);
  expect(toJson(tree)).toMatchSnapshot();
});

it('renders correctly empty config', () => {
  const config = {
    copyright: 'copyright',
    siteRss: "rss"
  };
  const tree = shallow(<Footer config={config} />);
  expect(toJson(tree)).toMatchSnapshot();
});
