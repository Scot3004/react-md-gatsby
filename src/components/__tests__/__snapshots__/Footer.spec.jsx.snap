// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly empty config 1`] = `
<footer
  className="footer"
>
  <div
    className="notice-container"
  >
    <div
      className="copyright"
    >
      copyright
    </div>
    <div
      className="rss"
    >
      <GatsbyLink
        to="rss"
      >
        <withInk(withTooltip(Button))
          flat={true}
          iconClassName="fa fa-rss"
          inkTransitionEnterTimeout={450}
          inkTransitionLeaveTimeout={300}
          inkTransitionOverlap={150}
          primary={true}
          swapTheming={true}
        >
          Subscribe
        </withInk(withTooltip(Button))>
      </GatsbyLink>
    </div>
    <div
      className="based-on"
    >
      Based on
      <a
        className="footer-link"
        href="https://github.com/Vagr9K/gatsby-material-starter"
      >
        Gatsby Material Starter
      </a>
      .
    </div>
  </div>
</footer>
`;

exports[`renders correctly fixedFooter 1`] = `
<footer
  className="footer footer-fixed"
>
  <div
    className="notice-container"
  >
    <div
      className="copyright"
    />
    <div
      className="rss"
    >
      <GatsbyLink
        to="rss"
      >
        <withInk(withTooltip(Button))
          flat={true}
          iconClassName="fa fa-rss"
          inkTransitionEnterTimeout={450}
          inkTransitionLeaveTimeout={300}
          inkTransitionOverlap={150}
          primary={true}
          swapTheming={true}
        >
          Subscribe
        </withInk(withTooltip(Button))>
      </GatsbyLink>
    </div>
    <div
      className="based-on"
    >
      Based on
      <a
        className="footer-link"
        href="https://github.com/Vagr9K/gatsby-material-starter"
      >
        Gatsby Material Starter
      </a>
      .
    </div>
  </div>
</footer>
`;

exports[`renders correctly userlinks 1`] = `
<footer
  className="footer"
>
  <UserLinks
    config={
      Object {
        "siteRss": "rss",
        "userLinks": "UserLinks",
      }
    }
    labeled={true}
  />
  <div
    className="notice-container"
  >
    <div
      className="copyright"
    />
    <div
      className="rss"
    >
      <GatsbyLink
        to="rss"
      >
        <withInk(withTooltip(Button))
          flat={true}
          iconClassName="fa fa-rss"
          inkTransitionEnterTimeout={450}
          inkTransitionLeaveTimeout={300}
          inkTransitionOverlap={150}
          primary={true}
          swapTheming={true}
        >
          Subscribe
        </withInk(withTooltip(Button))>
      </GatsbyLink>
    </div>
    <div
      className="based-on"
    >
      Based on
      <a
        className="footer-link"
        href="https://github.com/Vagr9K/gatsby-material-starter"
      >
        Gatsby Material Starter
      </a>
      .
    </div>
  </div>
</footer>
`;
